code/src/uart_nmea_c.cpp

```diff
@@ -0,0 +1,120 @@
+#include <uart_nmea_c.hpp>
+#include <cstdint>
+#include <cmath>
+#include <string.hpp>
+
+namespace r2d2::location {
+    uart_nmea_c::uart_nmea_c(r2d2::usart::usart_connection_c &usart_port)
+        : usart(usart_port) {
+    }
+
+    frame_coordinate_s uart_nmea_c::compress(const gga_s &source) {
+        return compress(source.longitude, source.latitude,
+                        (source.north_south_hemisphere == 'N'),
+                        (source.east_west_hemisphere == 'E'), source.altitude);
+    }
+
+    frame_coordinate_s uart_nmea_c::get_location() {
+        enum get_location_state {
+            wait_for_start,
+            get_message_id,
+            check_message_id,
+            get_sentence
+        };
+
+        get_location_state state = wait_for_start;
+        char uart_input = 0;
+
+        while (true) {
```

Comment:
this while loop should be removed

Reply:
We can probably keep it like this for now and fix it when the code reviewers want us to change this to master